org.eclipse.dawnsci.analysis.dataset.test/src/org/eclipse/dawnsci/analysis/dataset/StreamTest.java

```diff
@@ -0,0 +1,150 @@
+/*-
+ * Copyright (c) 2011, 2014, 2016 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Jacob Filik - initial API and implementation and/or initial documentation
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ */
+package org.eclipse.dawnsci.analysis.dataset;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.eclipse.january.DatasetException;
+import org.eclipse.january.dataset.Comparisons;
+import org.eclipse.january.dataset.Dataset;
+import org.eclipse.january.dataset.DatasetFactory;
+import org.eclipse.january.dataset.DatasetUtils;
+import org.eclipse.january.dataset.DoubleDataset;
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.january.dataset.Random;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * @see the full slicing unit tests in plugin    uk.ac.diamond.scisoft.analysis.test
+ *                                     package   uk/ac/diamond/scisoft/analysis/dataset
+ * @author Matthew Gerring
+ *
+ */
+public class StreamTest {
+
+	@Test
+	public void testLazyDatasetStream() {
+		Dataset original = DatasetFactory.createRange(DoubleDataset.class,10);
+		Dataset a = original.getSlice().reshape(new int[]{1,1,10});
+		a = DatasetUtils.tile(a, new int[]{4,5,1});
+		Stream<ILazyDataset> lzStream = Slices.stream(a, false, 0, 1);
+		doTest(lzStream.map(d -> safeSlice(d)), original);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. (sonar.java.source not set. Assuming 8 or greater.) [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1612)